src/pages/index.tsx

```diff
@@ -3,20 +3,110 @@ import { type NextPage } from "next";
 import { Layout } from "../components/Layout";
 import { useRouter } from "next/router";
 import dynamic from "next/dynamic";
-import { Posts } from "../components/Post";
+import { PostList } from "../components/PostList";
+import { api } from "../utils/api";
+import { useState } from "react";
+import { useDebounce } from "../hooks/useDebounce";
+import { Pagination } from "../components/Pagination";
+import { PostListFiltersValues, SearchPost } from "../components/SearchPost";
+import { z } from "zod";
+import { BlogPostCategory } from "../types/blogPost";
+import qs from "qs";
+
+const POST_LIST_FILTERS_DEBOUNCE = 500;
+const POSTS_PER_PAGE = 8;
+
+const queryParamsSchema = z.object({
+  search: z.string().optional().default(""),
+  categories: z.array(z.nativeEnum(BlogPostCategory)).optional().default([]),
+});
+
+interface QueryParams {
+  search?: string;
+  categories?: string;
+}
 
 const HomePage: NextPage = () => {
+  const [activePage, setActivePage] = useState(0);
+  const queryParams = qs.parse(window.location.search, {
+    ignoreQueryPrefix: true,
+  }) as QueryParams;
+  const [postListFilters, setPostListFilters] = useState<PostListFiltersValues>(
+    () =>
+      queryParamsSchema.parse({
+        categories: queryParams.categories?.split(","),
+        search: queryParams.search,
+      })
+  );
   const router = useRouter();
+
+  const debounce = useDebounce(
```

Comment:
`debounce` can be anything. You can name it `debouncedSearch` instead.

And to make things simpler, I think you can make whole filters debounced, so 

```
const deboucedPostListFilters = useDebounce(
    postListFilters,
    POST_LIST_FILTERS_DEBOUNCE
  );
```

And then

```

  const blogPostQuery = api.blogPost.getBlogPosts.useQuery({
    page: activePage,
    perPage: POSTS_PER_PAGE,
    categories: deboucedPostListFilters.categories,
    search: deboucedPostListFilters.search,
  });
```